### DISCUSSION TITLE
Please add appear offline feature

### DISCUSSION THREAD

**Discussion by MohammedNoureldin** [element-hq/element-meta#681] [Ideas]
The requested feature is out of the box from the title of this issue,
a lot of people I know (and me), would like to have `Appear offline or invisible` feature, so we are still to chat, but while appearing offline.

**Comment by uhoreg** [comment]
Related to the "Don't send presence" request in vector-im/element-web#660

**Comment by MohammedNoureldin** [comment]
May I help some how ti implement it? I searched a little bit yesterday where send presence API is being called, but I didn't find it, maybe you can tell me where it is being called, and I will implement this feature.

**Comment by t3chguy** [comment]
I'd imagine its sent from the bottom layer of the application, part of syncing in https://github.com/matrix-org/matrix-js-sdk

**Comment by MohammedNoureldin** [comment]
I know how to use that API, I did it yesterday using `curl` to set the status from offline to online for example, but what I mean is, where is it getting called in `Riot`, do you have any Idea about that?

**Comment by t3chguy** [comment]
Actually I think an implicit side effect of `/sync` may be a presence effect, I cannot remember if this is accurate or not though

**Comment by t3chguy** [comment]
I think the purpose is that if you set your presence to offline, Synapse will respect it and not update on a `/sync`

**Comment by MohammedNoureldin** [comment]
Sorry, what I did yesterday was a part from this:
http://matrix.org/docs/api/client-server/#!/Presence/put_matrix_client_r0_presence_userId_status

Could you tell me where does that happen in Riot project? what I use for example on my web server is just what I downloaded from Riot repository, nothing from that SDK.

**Comment by t3chguy** [comment]
@mne1991 its as simple as setting it to `offline` it seems:
![image](https://user-images.githubusercontent.com/2403652/28754416-b71fadae-753c-11e7-9047-a27e6fe0f04a.png)

**Comment by MohammedNoureldin** [comment]
Oh that seems good, I was actually looking here in this page:
http://matrix.org/docs/api/client-server/#!/Presence/put_matrix_client_r0_presence_userId_status

 where did you find that?

**Comment by t3chguy** [comment]
@mne1991 the specification of the protocol. http://matrix.org/docs/spec/client_server/r0.2.0.html#get-matrix-client-r0-sync

**Comment by lampholder** [comment]
Flagged feature p3 'cause this isn't on the core team's immediate roadmap, but it would be a great thing to have.

**Comment by MohammedNoureldin** [comment]
Yes, it would really be very helpful and great, it saves the privacy of users.

**Comment by superboum** [comment]
I am also interested by this feature.

**Comment by kittykat** [comment]
Moving this issue to discussions in Element meta as we need to make a cross platform decision on how to proceed :+1:

**Comment by Spacefish** [comment]
I think that discussion should be arround "presence" state in general.
As of today the matrix protocol knows 4 states:
- online
- unavailiable
- org.matrix.msc3026.busy (this is optional and only proposed see MSC3026)
- offline

as well as and optional "status_msg".. As far as i understand it, the status_msg should / can be used to indicate the time since the user is "unavailiable", by setting it to something like "since 14:12 today"

_Idle Detection_
Recently major browser (since Chrome 94) introduced the "Idle Detection API" this could be used to automatically set an "unavailiable" state after some inactivity or if the user locks the screen. See: https://developer.mozilla.org/en-US/docs/Web/API/Idle_Detection_API
This could play a role for element-web and the electron app and i guess it´s a heavily requested features by some users.

**Comment by Kaschi14** [comment]
Any updates on this?

**Comment by mca222** [comment]
The discussion started in 2017, so any news?
